engine/src/cubos/engine/tools/palette_editor/plugin.cpp

```diff
@@ -0,0 +1,104 @@
+#include <imgui.h>
+
+#include <cubos/core/data/old/debug_serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/renderer/plugin.hpp>
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+#include <cubos/engine/tools/palette_editor/plugin.hpp>
+#include <cubos/engine/voxels/plugin.hpp>
+
+using cubos::core::data::old::Debug;
+using cubos::core::ecs::EventReader;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using namespace cubos::engine;
+using tools::AssetSelectedEvent;
+
+struct SelectedPaletteInfo
+{
+    Asset<VoxelPalette> paletteAsset;
+    VoxelPalette paletteData; // so we dont local copy every frame, but only when palette is selected
+    bool active;
+    bool modified;
+    uuids::uuid activePaletteId;
+    std::pair<uint16_t, VoxelMaterial> modifiedMaterial;
```

Comment:
If I move this to system, does it not create it every time the system runs?

Reply:
If you mean the variable, it's perfectly fine, since its so small. My previous concern was with the voxel palette itself being copied, which is really big.